web/src/games/barnbarter/game.test.ts

```diff
@@ -0,0 +1,340 @@
+import { Client } from 'boardgame.io/client';
+import { Local } from 'boardgame.io/multiplayer';
+import { BarnBarterGame, _setup } from './game';
+import { Ctx } from 'boardgames.io';
+
+// https://github.com/nicolodavis/boardgame.io/blob/master/docs/testing.md
+it('Whole Game', () => {
```

Comment:
Would be nice to add smaller tests too, i.e. for the helper functions. This helps with debugging in case of a bug.

Reply:
Yes thats true. I'm not good at doing unit tests. I like the whole end-to-end, because it proves to me that all actions that the user can take work as expected. i'll add a lot of 'assertions' to make sure that if something breaks, its easy to see exactly where.

The problem with testing individual moves is that it always takes a lot of setup to reach the exact state you want to be at. E.g. a certain move is only possible very late in the game. So I just create a long long game going down all paths.